contracts/interchain-token-service/src/contract.rs

```diff
@@ -238,3 +328,381 @@ impl AxelarExecutableInterface for InterchainTokenService {
         event::executed(&env, source_chain, message_id, source_address, payload);
     }
 }
+
+#[cfg(test)]
+pub mod tests {
+    use super::*;
+    use crate::types::*;
+    use axelar_soroban_std::assert_ok;
+    use soroban_sdk::{testutils::Address as _, Address, Bytes, BytesN, Env, String};
+    use std::vec;
+    use std::vec::Vec;
+
+    const HUB_CHAIN: &str = "hub_chain";
+    const P2P_CHAIN: &str = "p2p_chain";
+    const HUB_ADDRESS: &str = "hub_address";
+
+    fn setup_env<'a>() -> (Env, InterchainTokenServiceClient<'a>) {
+        let env = Env::default();
+        let owner = Address::generate(&env);
+        let gateway = Address::generate(&env);
+        let gas_service = Address::generate(&env);
+        let contract_id = env.register(InterchainTokenService, (&owner, gateway, gas_service));
+        let client = InterchainTokenServiceClient::new(&env, &contract_id);
+
+        (env, client)
+    }
+
+    fn register_chains(env: &Env, client: &InterchainTokenServiceClient) {
+        env.mock_all_auths();
+
+        let chain = String::from_str(&env, P2P_CHAIN);
+        let addr = String::from_str(&env, "trusted_address");
+        client.set_trusted_address(&chain, &addr);
+
+        let chain = String::from_str(&env, HUB_CHAIN);
+        client.set_trusted_address(&chain, &client.its_hub_routing_identifier());
+
+        let chain = client.its_hub_chain_name();
+        let addr = String::from_str(&env, HUB_ADDRESS);
+        client.set_trusted_address(&chain, &addr);
+    }
+
+    fn bytes_from_hex(env: &Env, hex_string: &str) -> Bytes {
```

Comment:
This `bytes_from_hex` function is defined twice, the other one is in [abi.rs](https://github.com/axelarnetwork/axelar-cgp-soroban/blob/ce8d3af09bff15b85d94b5a2502806411a62374e/contracts/interchain-token-service/src/abi.rs#L257).

Could we move this to test utils to avoid defining the same function twice?